### ISSUE TITLE
Edit the shipping price

### ISSUE THREAD

**Issue by madsoliver** [PrestaShop/PrestaShop#17225]
#### Is your feature request related to a problem?
Sometimes we want to create an order in BO and we can't use any shipping price which is defined in BO, we've to use an individual shipping price. But it is not possible to edit the shipping price in BO.

#### Describe the solution you'd like
The shipping price should be possible to edit and change.

#### Additional context
BO > Orders > Orders > detail info > section Shipping > Shipping cost

![Screenshot 2020-01-17 13 35 47](https://user-images.githubusercontent.com/30602638/72614924-c5086b00-3933-11ea-94f2-fbc83f06ec54.png)

Edit: To respect the template of the features

**Comment by khouloudbelguith** [comment]
Hi @onklmaps,

Thanks for your report.
Ping Thanks for your report.
Ping @PrestaShop/prestashop-product-team @matks what do you think of this improvement?

Thanks!

**Comment by Hlavtox** [comment]
Yeah, that was missing and we always used a module to add this functionality.

We need/would be nice:
- Change shipping price
- Change payment type
- Resend order confirm
- Change customer (not only edit the order's one)

Already have:
- Change carrier
- Change discounts
- Change products
- Change addresses

**Comment by PeNov** [comment]
Hello,
I agree with Hlavtox, such functions would be great.

**Comment by madsoliver** [comment]
Common problem: customer miss-spelled email, want to change, then get the order_conf again.

**Comment by colinegin** [comment]
@MatShir what do you think for a future improvement of the order page ?

**Comment by hibatallahAouadni** [comment]
Ping @PrestaShop/product-team what's the status of this issue ? is it **`To Do`** ?

**Comment by marionf** [comment]
@hibatallahAouadni, no this need to be specified

**Comment by madsoliver** [comment]
Is any input needed from me?

**Comment by skellatore** [comment]
Desperately need this feature.  What's worse Is if you want to add the tracking number to a "Remote" generated shipping price, the carrier is no longer in the list!
Often I get customers that forget items and want to "combine" shipping resulting in a lower rate than what was charged across two orders - I deliver a in-store credit with a voucher - this is well received by the customer and drives repeat business.
And finally, Why can't a voucher be used to pay for shipping! how ridiculous! What if someone buys a gift-voucher for someone else - they have to pay for shipping even if it falls within the value of the gift-voucher purchase!

**Comment by boo-code** [comment]
Still absent on 9.2.x: `update_shipping_modal.html.twig` renders only the tracking-number and carrier widgets (the remainder goes through `form_rest` into a `d-none` div), and `UpdateOrderShippingDetailsCommand` carries `orderId`, `currentOrderCarrierId`, `newCarrierId`, `trackingNumber`. No cost field anywhere on that path.

What this thread has not recorded is that the persistence half already exists, and it postdates the "we always used a module" observation. `OrderAmountUpdater::updateOrderTotals()` recomputes shipping from `$cart->getOrderTotal(..., Cart::ONLY_SHIPPING, ...)`, but when `PS_ORDER_RECALCULATE_SHIPPING` is off it restores the order's stored `total_shipping_tax_incl/excl` and rebalances `total_paid*` (`total_discounts*` on a free-shipping order); `updateOrderCarrier()` then propagates them into `ps_order_carrier.shipping_cost_tax_incl/excl`. A manually set shipping total is therefore already durable across later order edits, in that mode only.

That setting ships as `1` (`install-dev/data/xml/configuration.xml:843`), so under defaults an entered cost is recomputed away on the next product add, removal or quantity change. This is the specification gap, and it is a data-model question rather than a form-field one: the setting is shop-scoped, so making an edited cost independent of it, or forcing it off for one order, both require per-order state. Invoiced orders pose the same question against a document already issued.
